### ISSUE TITLE
Add Types!

### ISSUE THREAD

**Issue by bh2smith** [pinnaculum/py-multiformats-cid#1]
Mypy complains and I have to silence it like this (which is not ideal).

```
from multiformats_cid import from_bytes  # type: ignore
```

If the project is already typed, then it should be as easy as adding an empty file called `py.typed` to the base module.

and adding this to setup configuration:

```
[options.package_data]
cid = py.typed
```

It appears I posted the issue in the wrong place:

https://github.com/multiformats/cid/issues/56

**Comment by PancakesArchitect** [comment]
Hi @bh2smith. I recently forked [ipld/py-cid](https://github.com/ipld/py-cid) and fixed the issues with the python requirements, wasn't aware of this problem, i'll post a fix asap, thank you.

**Comment by PancakesArchitect** [comment]
Following your recommendation, i've updated the *setup.cfg* file, tested it with mypy, and indeed the errors go away. Published v0.4.2 to pypi. Thanks !

**Comment by bh2smith** [comment]
This is great news, however it is not quite right:

```
Successfully installed py-multiformats-cid-0.4.2

$ mypy pysrc --strict

fetch/ipfs.py:7: error: Skipping analyzing "multiformats_cid": module is installed, but missing library stubs or py.typed marker
fetch/ipfs.py:7: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

You actually need to create an empty file called `py.typed`

based on your previous commit https://github.com/pinnaculum/py-multiformats-cid/commit/547ad4f4c60847aa032d4d388391e5f295d8c785, It looks like the only thing left to do is add the file.

```
$ touch multiformats_cid/py.typed
```


Update, it was much more involved than that... I forked the project and did the things in #2

**Comment by PancakesArchitect** [comment]
@bh2smith Thank you so much for this, the typing really improves the module a lot. Reviewing your PR now, will merge it and list you in the contributors. I'll also add a CI workflow to run the unit tests and will publish to pypi after that.

**Comment by bh2smith** [comment]
> @bh2smith Thank you so much for this, the typing really improves the module a lot. Reviewing your PR now, will merge it and list you in the contributors. I'll also add a CI workflow to run the unit tests and will publish to pypi after that.

Another thing that could be cool to do is auto publish on tagged releases. I could link you to an example of how I do it in a few projects. It's a bit awkward to setup at first but totally worth it in the end. 

https://github.com/cowprotocol/dune-client/blob/main/.github/workflows/py-publish.yaml

Something like this workflow and possibly a couple changes to setup files.

**Comment by PancakesArchitect** [comment]
> > @bh2smith Thank you so much for this, the typing really improves the module a lot. Reviewing your PR now, will merge it and list you in the contributors. I'll also add a CI workflow to run the unit tests and will publish to pypi after that.
> 
> Another thing that could be cool to do is auto publish on tagged releases. I could link you to an example of how I do it in a few projects.

Merged your PR. Yeah the auto-publish with twine is no problem, just reused a workflow i had, will be ready soon.

Edit: thanks for the workflow link, might be more elegant by reusing an action.

**Comment by PancakesArchitect** [comment]
Question: is the *from __future__ import annotations* import strictly necessary ? This breaks support for python 3.6, but that's probably an acceptable choice.

**Comment by bh2smith** [comment]
> Question: is the *from __future__ import annotations* import strictly necessary ? This breaks support for python 3.6, but that's probably an acceptable choice.

Yes actually this implementation as is will only work for >= 3.8

It can be adjusted to also work with 3.7, but I can't go as far back as 3.6

See here

https://github.com/pinnaculum/py-multiformats-cid/pull/2#issuecomment-1320875232


UPDATE: here is a PR for 3.7 compatibility
https://github.com/pinnaculum/py-multiformats-cid/pull/3

**Comment by PancakesArchitect** [comment]
Ok, will merge #3 then cause 3.7 can't be considered a dinosaur yet. Published v0.4.3 to pypi.
